platform/security/core/security-core-services/src/main/java/ddf/security/soap/impl/SecureProxyServiceFactoryImpl.java

```diff
@@ -155,17 +162,19 @@ private SecurityToken getSecurityToken(WebServiceProperties wsp, Serializable se
     configureProxyFactoryProperties(clientFactory, token, wsp);
     clientFactory.getOutInterceptors().add(new TokenPassThroughInterceptor());
 
-    ProxyServiceType proxyServiceType;
-    try {
-      proxyServiceType = clientFactory.create(wsp.serviceClass);
-    } catch (ServiceConstructionException e) {
-      LOGGER.debug("Unable to use WSDL to build client. Attempting to use service class.", e);
-      unavailableWsdls.add(wsp.endpointWsdlURL);
-      clientFactory.getClientFactoryBean().getServiceFactory().setPopulateFromClass(true);
-      proxyServiceType = clientFactory.create(wsp.serviceClass);
-    }
-
-    return proxyServiceType;
+    return AccessController.doPrivileged(
+        (PrivilegedAction<ProxyServiceType>)
+            () -> {
+              try {
+                return clientFactory.create(wsp.serviceClass);
+              } catch (ServiceConstructionException e) {
+                LOGGER.debug(
+                    "Unable to use WSDL to build client. Attempting to use service class.", e);
+                unavailableWsdls.add(wsp.endpointWsdlURL);
+                clientFactory.getClientFactoryBean().getServiceFactory().setPopulateFromClass(true);
+                return clientFactory.create(wsp.serviceClass);
+              }
+            });
```

Comment:
❓ In the future, do we plan to check a custom permission before doing this?

Reply:
Yes, I'll probably add something just like I did for the REST client.